identity/src/main/java/com/stripe/android/identity/ml/IDDetectorAnalyzer.kt

```diff
@@ -0,0 +1,165 @@
+package com.stripe.android.identity.ml
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Rect
+import android.util.Log
+import com.stripe.android.camera.CameraPreviewImage
+import com.stripe.android.camera.framework.Analyzer
+import com.stripe.android.camera.framework.AnalyzerFactory
+import com.stripe.android.camera.framework.image.cropCameraPreviewToSquare
+import org.tensorflow.lite.DataType
+import org.tensorflow.lite.Interpreter
+import org.tensorflow.lite.support.common.ops.NormalizeOp
+import org.tensorflow.lite.support.image.ImageProcessor
+import org.tensorflow.lite.support.image.TensorImage
+import org.tensorflow.lite.support.image.ops.ResizeOp
+import java.io.FileInputStream
+import java.nio.channels.FileChannel
+
+/**
+ * Result category of IDDetector
+ */
+enum class Category { NO_ID, PASSPORT, ID_FRONT, ID_BACK, INVALID }
+
+/**
+ * Result bounding box coordinates of IDDetector, in percentage values with regard to original image's width/height
+ */
+data class BoundingBox(
+    val top: Float,
+    val left: Float,
+    val width: Float,
+    val height: Float,
+)
+
+/**
+ * Analyzer to run a model input.
+ * TODO(ccen): reimplement with ImageClassifier
+ */
+internal class IDDetectorAnalyzer(context: Context) :
+    Analyzer<IDDetectorAnalyzer.Input, IDDetectorAnalyzer.State, IDDetectorAnalyzer.Output> {
+
+    private val tfliteInterpreter = Interpreter(
+        context.assets.openFd(modelName).use { fileDescriptor ->
```

Comment:
consider using a `TensorFlowLiteAnalyzerFactory` with a `ResourceFetcher` https://github.com/stripe/stripe-android/blob/master/stripecardscan/src/main/java/com/stripe/android/stripecardscan/framework/Fetcher.kt#L136

Reply:
Similar to the reason above, `ImageClassifier` has an [constructor](https://github.com/tensorflow/tflite-support/blob/fbfcb9930c41c547b237c58b3b6fb5f48e6d8b26/tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision/classifier/ImageClassifier.java#L130) that takes the model path as parameter. The interface&parameters left here is intentionally matching that interface.